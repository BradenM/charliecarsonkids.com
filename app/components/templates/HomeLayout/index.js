/**
 *
 * HomeLayout
 *
 * Templates
 */

import Text from 'atoms/Text';
import Banner from 'molecules/Banner';
import Footer from 'organisms/Footer';
import Header from 'organisms/Header';
import PropTypes from 'prop-types';
import React from 'react';

const HomeLayout = ({ children, bannerOptions }) => (
  <section className="homepage--wrapper h-100 w-100">
    {bannerOptions && (
      <Banner>
        <Text className="m-0 p-1" color="body">
          {bannerOptions.text}
        </Text>
      </Banner>
    )}
    <Header />
    {children}
    <Footer />
  </section>
);

HomeLayout.defaultProps = {
  bannerOptions: false,
};

HomeLayout.propTypes = {
  children: PropTypes.element,
  bannerOptions: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.shape({
      text: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
      color: PropTypes.string,
    }),
  ]),
};

export default HomeLayout;
