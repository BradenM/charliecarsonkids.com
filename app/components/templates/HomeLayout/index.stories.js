/**
 *
 * Stories for HomeLayout
 *
 * Templates
 */

import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import React from 'react';
import StoryRouter from 'storybook-react-router';
import { PlaceholderURL } from 'utils/getStatic';
import HomeLayout from '.';

export default {
  title: 'Design|Templates/HomeLayout',
  default: HomeLayout,
  decorators: [StoryRouter()],
  parameters: {
    centered: { disable: true },
  },
};

export const withPlaceholders = () => <HomeLayout />;

export const withDynamicVariables = () => {
  const bannerOpts = {
    text: text('Banner Text', 'banner text'),
    color: text('Banner Color', '#cecece'),
  };
  const heroOpts = {
    onButtonClick: action('onButtonClick'),
    text: text('Hero Text', 'Hero Title'),
    buttonText: text('Button Text', 'Click me'),
    imageSrc: text('Image Source', PlaceholderURL('1200')),
  };
  return (
    <HomeLayout bannerOptions={bannerOpts} heroOptions={heroOpts}>
      <div></div>
    </HomeLayout>
  );
};
