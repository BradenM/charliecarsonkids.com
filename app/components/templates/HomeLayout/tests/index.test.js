/**
 *
 * Tests for HomeLayout
 *
 * Templates
 */

// import '@testing-library/jest-dom/extend-expect'; // add some helpful assertions
import { render } from '@testing-library/react';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from '../../../../configureStore';
import HomeLayout from '../index';

const renderComponent = (store, history, props = {}) =>
  render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <HomeLayout {...props} />
      </ConnectedRouter>
    </Provider>,
  );

describe('<HomeLayout />', () => {
  let store;
  let history;

  beforeAll(() => {
    history = createMemoryHistory();
    store = configureStore({}, history);
  });
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    renderComponent(store, history);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderComponent(store, history);
    expect(firstChild).toMatchSnapshot();
  });
});
