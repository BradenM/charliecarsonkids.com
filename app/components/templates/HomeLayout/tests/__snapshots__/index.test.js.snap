// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeLayout /> Should render and match the snapshot 1`] = `
<div
  class="homepage-container container-fluid"
>
  <div
    class="row"
  >
    <div
      class="p-0 col"
    >
      <div
        class="Banner__BannerContainer-sc-1in81vp-0 hQEDkc p-0"
      >
        <div
          class="banner-wrapper py-1"
        >
          <div
            class="banner-text text-center"
          >
            <p
              class=" text-dark m-0 p-1"
            >
              banner content
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="d-flex justify-content-start px-3 py-2 col"
          >
            <div
              class="border-0  list-group list-group-horizontal"
            >
              <div
                class="IconGroup__BorderlessItem-sc-1br5l8c-0 igufRO list-group-item"
              >
                <div
                  class="IconButton__IconContainer-pjjsvs-0 bKDZXR"
                >
                  <div>
                    <div>
                      <div
                        data-src="https://static.charliecarsonkids.com/assets/icons/facebook-100.svg"
                        src=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="IconGroup__BorderlessItem-sc-1br5l8c-0 igufRO list-group-item"
              >
                <div
                  class="IconButton__IconContainer-pjjsvs-0 bKDZXR"
                >
                  <div>
                    <div>
                      <div
                        data-src="https://static.charliecarsonkids.com/assets/icons/instagram-100.svg"
                        src=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="IconGroup__BorderlessItem-sc-1br5l8c-0 igufRO list-group-item"
              >
                <div
                  class="IconButton__IconContainer-pjjsvs-0 bKDZXR"
                >
                  <div>
                    <div>
                      <div
                        data-src="https://static.charliecarsonkids.com/assets/icons/mail-100.svg"
                        src=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-end px-3 py-2 col"
          >
            <div
              class="border-0  list-group list-group-horizontal"
            >
              <div
                class="IconGroup__BorderlessItem-sc-1br5l8c-0 igufRO list-group-item"
              >
                <div
                  class="IconButton__IconContainer-pjjsvs-0 bKDZXR"
                >
                  <div
                    class="Anicon__HoverContainer-sc-119yqeq-0 iCLpQU anicon-container"
                  >
                    <div
                      style="width: 40px; height: 100%; transform: rotate(90deg); cursor: pointer;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="IconGroup__BorderlessItem-sc-1br5l8c-0 igufRO list-group-item"
              >
                <div
                  class="IconButton__IconContainer-pjjsvs-0 bKDZXR"
                >
                  <div
                    class="Anicon__HoverContainer-sc-119yqeq-0 iCLpQU anicon-container"
                  >
                    <div
                      style="width: 40px; height: 100%; transform: rotate(0deg); cursor: pointer;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="justify-content-md-center row"
        >
          <div
            class="d-flex justify-content-center py-3 col-md-auto"
          >
            <div
              class="w-50"
            >
              <img
                class="img-fluid"
                src="IMAGE_MOCK"
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <nav
              class="Navigation__StyledNavbar-sc-11o2ur6-0 bXviVH navbar navbar-expand-lg navbar-light"
            >
              <button
                aria-controls="responsive-navbar-nav"
                aria-label="Toggle navigation"
                class="navbar-toggler collapsed"
                type="button"
              >
                <span
                  class="navbar-toggler-icon"
                />
              </button>
              <div
                class="navbar-collapse collapse"
                id="responsive-navbar-nav"
              >
                <div
                  class="m-auto justify-content-around navbar-nav"
                >
                  <a
                    class="nav-link"
                    data-rb-event-key="new"
                    href="new"
                  >
                    new
                  </a>
                  <a
                    class="nav-link"
                    data-rb-event-key="boys"
                    href="boys"
                  >
                    boys
                  </a>
                  <a
                    class="nav-link"
                    data-rb-event-key="girls"
                    href="girls"
                  >
                    girls
                  </a>
                  <a
                    class="nav-link"
                    data-rb-event-key="baby"
                    href="baby"
                  >
                    baby
                  </a>
                  <a
                    class="nav-link"
                    data-rb-event-key="collections"
                    href="collections"
                  >
                    collections
                  </a>
                  <a
                    class="nav-link"
                    data-rb-event-key="about"
                    href="about"
                  >
                    about
                  </a>
                </div>
              </div>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="HomeHero__HeroContainer-sc-18ez42y-0 lcxRvI container"
      >
        <div
          class="Hero__StyledJumbotron-sc-1ele3ck-0 eBBpFn hero jumbotron"
        >
          <div
            class="h-100 container"
          >
            <div
              class="row"
            >
              <div
                class="pb-3 my-auto col-6"
              >
                <h1
                  class="Title-sc-3abvrr-0 Hero__HeroTitle-sc-1ele3ck-1 bqYlOi mb-0 font-weight-bold display-1 mb-0 font-weight-bold"
                  color="#fefefe"
                  size="1"
                >
                  HERO TITLE
                </h1>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="py-3 col-3"
              >
                <button
                  class="btn btn-outline-light btn-lg"
                  type="button"
                >
                  Click Me
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
